### PR TITLE
feat: common builds for framework plugins

### PR DESCRIPTION
**Pull request by dominikg**
work in progress for now.

cc @brillout @sapphi-red 

if you have ideas how we could make this more streamlined i'm eager to hear them.
Maybe we should extend the vite build to include these by default of have a new run option ?

### CODE REVIEW

**Comment by patak-dev** [comment]
Maybe we should consider specific plugins as "core plugins", and include support by default. Starting with all the framework plugins, not only Vue and React. We have a harcoded list of these plugins, and we check the package json of the tested project for each, and we do a clone/build/override for each like you do here. If necessary we could give an option to opt-out for certain exceptional cases, but I would assume that most users would like to test things against the latest of each plugin.

**Comment by brillout** [comment]
I think I lack the context to provide useful feedback, but LGTM 👍.

File: tests/iles.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ export async function test(options: RunOptions) {
 	await runInRepo({
 		...options,
 		repo: 'ElMassimo/iles',
+		overrides:{
+			"@vitejs/plugin-vue": true
+		},
```
Comment (patak-dev) [inline tests/iles.ts:10]: Why is this needed? Is it because the dependency isn't at the top-level package.json?

**Review by patak-dev — APPROVED** [review]

**Comment by patak-dev** [comment]
This is awesome 🔥

File: tests/iles.ts (hunk 1 above)
Comment (dominikg) [reply tests/iles.ts:10]: yes, we only scan the top level package.json. in monorepos you have to tell us you want/need it

Comment (patak-dev) [reply tests/iles.ts:10]: Gotcha
